library/std/src/sys/mod.rs

```diff
@@ -38,7 +38,7 @@ cfg_if::cfg_if! {
     } else if #[cfg(target_os = "hermit")] {
         mod hermit;
         pub use self::hermit::*;
-    } else if #[cfg(target_os = "wasi")] {
+    } else if #[cfg(any(target_os = "wasi", target_os = "wasi-preview2"))] {
         mod wasi;
         pub use self::wasi::*;
```

Comment:
One thing I might recommend here instead is to create a `wasi_preview2` module which copies `wasi/mod.rs` and uses `#[path = "..."]` to point to all the WASI files, making it clear that preview2 will be separate in the future but for now it's reusing all of wasi

Reply:
Oh yeah, using `#[path]` makes sense instead of copying.

Reply:
Gave this a try - let me know what you think